packages/backend/src/modules/tvl/modules/ChainModule.ts

```diff
@@ -86,26 +86,32 @@ function createChainModule(
   }
   logger = logger.tag(chain)
 
-  const blockNumberProviderConfig =
-    chainConfig.config.blockTimestampClientConfig
-  const blockTimestampClient = blockNumberProviderConfig
-    ? blockNumberProviderConfig.type === 'etherscan'
-      ? peripherals.getClient(EtherscanClient, {
-          apiKey: blockNumberProviderConfig.etherscanApiKey,
-          url: blockNumberProviderConfig.etherscanApiUrl,
-          chainId: chainConfig.config.chainId,
-        })
-      : peripherals.getClient(BlockscoutClient, {
-          url: blockNumberProviderConfig.blockscoutApiUrl,
-          chainId: chainConfig.config.chainId,
-        })
-    : undefined
-
   const rpcClient = peripherals.getClient(RpcClient, {
     url: chainConfig.config.providerUrl,
     callsPerMinute: chainConfig.config.providerCallsPerMinute,
   })
 
+  const options =
+    chainConfig.config.blockTimestampClientConfig === undefined
+      ? undefined
+      : chainConfig.config.blockTimestampClientConfig.type === 'etherscan'
+        ? {
+            type: 'Etherscan' as const,
+            apiKey:
+              chainConfig.config.blockTimestampClientConfig.etherscanApiKey,
+            url: chainConfig.config.blockTimestampClientConfig.etherscanApiUrl,
+            maximumCallsForBlockTimestamp: 3,
```

Comment:
SHould we use `env` for this?

Reply:
in theory we can but in practice I not see this one as being changed frequently, we already have a lot of envs, I am afraid this will introduce not needed noise